### DISCUSSION TITLE
Context discussion regarding click events inside Dialog

### DISCUSSION THREAD

**Discussion by mochadwi** [ReactiveCircus/FlowBinding#117] [Q&A]
Context Previous: https://github.com/ReactiveCircus/FlowBinding/issues/116#issuecomment-742972662

> How are you collection the flow? do you have a single collector in your activity or do you have another one in your BottomSheetDialogFragment?

we're using default BottomSheetDialogFragment (with custom layout), and setup the dialog with dynamic inflation, e.g:

```
fun showDialog() {
  val dialog = BottomSheetDialog(/* etc */)
  dialog.setContentView(dialogBinding.root)
  dialog.show()
}
```
we have multiple collector in our activity (we're using MVI same references from [Ezike here](https://github.com/Ezike/Baking-App-Kotlin/blob/master/features/recipes/recipe/src/main/java/com/example/eziketobenna/bakingapp/recipe/ui/RecipeFragment.kt#L75)), but no flow collector in dialog

> Each fragment and activity has its own LifecycleScope so sharing can be tricky. In this case does it make sense to communicate between your dialog and the host activity (assuming that's what you're trying to achieve) with a SharedFlow which is injected to both?

Agreed, we're trying to communicate between dialog and activity seamlessly

Ah, I see. Lemme clarify your statement with code example.

(1) we need to customize our own dialog/bottom dialog

```
class FlowDialog : BottomSheetFragment {
  // prepare the custom layout binding here
  @Inject
  lateinit var sharedFlow: SharedFlow
}
```

or maybe any references we can use

@ychescale9

**Comment by ychescale9** [comment]
Yeah something along the line. Ideally this instance is scoped to the activity and any of its child fragments.

Btw if you use jetpack navigation there’s a new [NavBackStackEntry API](https://developer.android.com/guide/navigation/navigation-programmatic) which exposes LiveData. Might be another option if you that works for your taste.
